package/kedro_viz/server.py

```diff
@@ -317,9 +318,22 @@ def format_pipelines_data(pipelines: Dict[str, "Pipeline"]) -> Dict[str, list]:
     # keep track of node_id -> set(child_node_ids) for layers sorting
     node_dependencies = defaultdict(set)
     tags = set()
+    # keep track of modular pipelines
+    modular_pipelines = set()
 
     for pipeline_key, pipeline in pipelines.items():
         pipelines_list.append({"id": pipeline_key, "name": _pretty_name(pipeline_key)})
+        modular_pipelines.update(
```

Comment:
This will include the modular pipelines from all pipelines, but we actually need only the ones from the currently selected pipeline.

Reply:
When viz first fetches data about the pipelines and nodes etc it will call this `format_pipelines_data`. When this call is made, the default pipeline will be the selected pipeline. As far as I understand we don't do any filtering at this point, so the pipeline data returned here contains everything. Subsequent calls will then go to the `/api/pipelines/<string:pipeline_id>` endpoint, where indeed the data will be filtered to only return details for the selected data.